### ISSUE TITLE
Can Edge Bate be used as an alternative when distributing WebView2?

### ISSUE THREAD

**Issue by sln162** [MicrosoftEdge/WebView2Feedback#2903]
In China, more than 30% of users are still using Win7, and there are some systems with devices lower than Win10-2004. In addition, when the system update is turned off when the system is larger than 2004, WebView2 still needs to be installed in these environments. In order to prevent installation failures (there will be some strange Gosht systems that have been modified), can I use Edge Bate as an alternative? Or should the fixed version be an alternative? Or only the runtime installation is provided. If the installation fails, can I send a feedback request for your assistance?

Is there any official suggestion? thank you

**Comment by LiangTheDev** [comment]
Why would Edge Beta be better/easier to install than Edge WebView2 Runtime? If Edge Beta is on the machine but WebView2 Runtime is not, we will use Edge Beta to run WebView2. This happens automatically. GetAvailableCoreWebView2BrowserVersionString API actually will return the version of Edge Beta in this case.

**Comment by sln162** [comment]
@LiangTheDev  Yes, it just provides more possibilities. However, I found a problem when using the command: Microsoft Edge Webview2Setup Exe/install. After the installation is completed, the system will automatically exit. However, MicrosoftEdgeWebView2RuntimeInstallerX64.exe/install does not exit. It will prompt that the installation is complete and the user needs to click to close. This is not convenient for us to develop, because my program needs to wait for the end of the process to continue.

**Comment by LiangTheDev** [comment]
Have you used /silent command line switch for MicrosoftEdgeWebview2Setup.exe or MicrosoftEdgeWebView2RuntimeInstallerX64.exe? That should perform the installation silently and does not require user click anything. See https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/distribution for more details.
FWIW, install of Edge browser (beta channel or not) actually would be more impactful, it automatically starts the browser and creates start menu entry, etc.

**Comment by sln162** [comment]
@LiangTheDev  My idea is that I don't need/silent, but only use/install to let users see the installation program, so that they can feed back to me when they encounter errors. However, the two programs are different for/install. When the installation of MicrosoftEdgeWebView2RuntimeInstallerX64.exe is completed, there will be a prompt, which requires the user to click. Is there any other command that can remove this prompt?

Edge Beta is really not suitable, and it is easier for users to uninstall it, so it will not be considered for the time being.

**Comment by sln162** [comment]
The advantage of not choosing/silent is that you can see that the source of the program is Microsoft, you can see the download progress, and users can know what is going on when blocked by some security software. I have built a separate form for it. If it is/silent, it seems simpler. I just need to embed it in the program and/silent start the installation.

**Comment by LiangTheDev** [comment]
Unfortunately, there is currently no command line switch to show the installation UI but skip the completion dialog. You would have to build separate UI stating that you are installing Edge WebView2 Runtime, and then dismiss it when done. The Edge update/installation logs should be the best source of information if install error happened.

**Comment by LiangTheDev** [comment]
I am creating an internal bug for this. For not, please work around it.

**Comment by sln162** [comment]
@LiangTheDev  I think it's a bug. You can pop up a prompt without the/install command. Once you use it, you should not prompt, otherwise you need to give another command to mask the prompt (not yet)
